portal/models/user.py

```diff
@@ -44,60 +44,23 @@
     TRUENTH_ID, TRUENTH_USERNAME) + TRUENTH_PROVIDER_SYSTEMS
 
 
-class Extension:
-    """Abstract base class for common user extension FHIR objects"""
-    __metaclass__ = ABCMeta
-
+class UserIndigenousStatusExtension(CCExtension):
+    # Used in place of us-core-race and us-core-ethnicity for
+    # Australian configurations.
     def __init__(self, user, extension):
         self.user, self.extension = user, extension
 
-    @abstractproperty
-    def children(self):  # pragma: no cover
-        pass
-
-    def as_fhir(self):
-        if self.children.count():
-            return {
-                'url': self.extension_url,
-                'valueCodeableConcept': {
-                    'coding': [c.as_fhir() for c in self.children]}
-            }
-
-    def apply_fhir(self):
-        assert self.extension['url'] == self.extension_url
-        # track current concepts - must remove any not requested
-        remove_if_not_requested = {e.code: e for e in self.children}
-
-        for coding in self.extension['valueCodeableConcept']['coding']:
-            try:
-                concept = Coding.query.filter_by(
-                    system=coding['system'], code=coding['code']).one()
-            except NoResultFound:
-                raise ValueError("Unknown code: {} for system{}".format(
-                                     coding['code'], coding['system']))
-            if concept.code in remove_if_not_requested:
-                # The concept existed before and is to be retained
-                remove_if_not_requested.pop(concept.code)
-            else:
-                # Otherwise, it's new; add it
-                self.children.append(concept)
-
-        # Remove the stale concepts that weren't requested again
-        for concept in remove_if_not_requested.values():
-            self.children.remove(concept)
-
-
-class UserIndigenousStatusExtension(Extension):
-    # Used in place of us-core-race and us-core-ethnicity for
-    # Australian configurations.
     extension_url = TRUENTH_EXTENSTION_NHHD_291036
 
     @property
     def children(self):
         return self.user.indigenous
 
 
-class UserEthnicityExtension(Extension):
+class UserEthnicityExtension(CCExtension):
+    def __init__(self, user, extension):
```

Comment:
I wasn't sure the best way to approach the `init` method. When Extensions are used in the user model, the method takes a `user` parameter, but for orgs we want it to take an `organization` instead. So I just put it in the extended classes, instead of the base class.

I'm definitely open to suggestions on that one, if there's a good way to work it into the base class.